logviewer/src/main/java/org/togetherjava/logwatcher/config/Config.java

```diff
@@ -0,0 +1,122 @@
+package org.togetherjava.logwatcher.config;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.context.annotation.Configuration;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Basic Class for accumulating Information which is required to start this application
+ */
+@Configuration
+public class Config {
+
+    private static final AtomicReference<String> CONFIG_PATH = new AtomicReference<>();
+
+    public static void init(final String pathToConfig) {
+        if (!CONFIG_PATH.compareAndSet(null, pathToConfig)) {
+            throw new IllegalStateException(
+                    "Config Path already set to %s".formatted(CONFIG_PATH.get()));
+        }
+    }
+
+
+    /**
+     * Client-Name of the OAuth2-Application
+     */
+    private final String clientName;
+
+    /**
+     * Client-ID of the OAuth2-Application
+     */
+    private final String clientId;
+
+    /**
+     * Client-Secret of the OAuth2-Application
+     */
+    private final String clientSecret;
+
+    /**
+     * Discord-Username of the User to add as 1. User
+     */
+    private final String rootUserName;
+
+    /**
+     * Discord-ID of the User to add as 1. User
+     */
+    private final String rootDiscordID;
+
+    /**
+     * Path of the Log directory
+     */
+    private final String logPath;
+
+    /**
+     * Redirect Path for OAuth2
+     */
+    private final String redirectPath;
+
+    /**
+     * Path for this Database
+     */
+    private final String databasePath;
+
+    public Config() {
+        final JsonNode jsonNode = getJsonNode();
+
+        this.clientName = jsonNode.get("clientName").asText();
+        this.clientId = jsonNode.get("clientId").asText();
+        this.clientSecret = jsonNode.get("clientSecret").asText();
+        this.rootUserName = jsonNode.get("rootUserName").asText();
+        this.rootDiscordID = jsonNode.get("rootDiscordID").asText();
+        this.logPath = jsonNode.get("logPath").asText();
+        this.redirectPath = jsonNode.get("redirectPath").asText();
+        this.databasePath = jsonNode.get("databasePath").asText();
+    }
+
+    private JsonNode getJsonNode() {
+        final String pathToConfig = Objects.requireNonNull(CONFIG_PATH.get(), "Path not Set");
+        try {
+            return new ObjectMapper().readTree(Path.of(pathToConfig).toFile());
```

Comment:
Prefer having the `ObjectMapper` pre-created, maybe as a field.

Reply:
Or Autoinjected by Spring? Would that also be fine?

Reply:
Did you use dependency injection elsewhere in the program already? If you did not use it much yet, I would prefer to not introduce it just for this mini occasion to keep complexity and _black magic_ low for a beginner.

If you already made heavy usage of it anyways, I am completely fine with having it again. I am the last who blocks people writing their code the way they love in a community project like this.

Reply:
I pretty much used it in in every Class